examples/jsm/csm/CSM.js

```diff
@@ -200,14 +204,20 @@ export class CSM {
 
 		const camera = this.camera;
 		const frustums = this.frustums;
+
+		// for each frustum we need to find its min-max box aligned with the light orientation
+		// the position in _lightOrientationMatrix does not matter, as we transform there and back
+		_lightTarget.copy( this.lightDirection ).add( camera.position );
+		_lightOrientationMatrix.lookAt( camera.position, _lightTarget, _up );
```

Comment:
Instead of defining a custom `up` vector, could you use `light.up` instead?

Reply:
Good idea. The trouble is `light` is not available at the point where `_lightOrientationMatrix` is computed, which is outside the loop. I can use `camera.up`, though - adding a commit doing it that way.

Reply:
Although using `camera.up` isn't right. It has to be the `up` vector of the light. Probably better to move the computation inside the loop where you can access `light`.

Reply:
This may sound like a reasonable thing as first sight, but at second side that would achieve nothing more than my original code. The `light` inside of the loop is a synthetic light created by CSM plugin (see `createLights`). It has no relation to the real directional light in the scene, one could use `Object3D.DefaultUp` instead. The light direction is passed in `data.lightDirection` and I have no access to the `DirectionalLight` object. I could pass the `up` next to the direction, or perhaps the whole interface could be changed to pass the whole light object.

Reply:
Note: previous implementation effectively used `Object3D.DefaultUp`, as the light direction is set using `shadowCamera.lookAt( _lookTarget );` in `LightShadow.matrices`. `DirectionalLightShadow` uses `new OrthographicCamera` to construct its camera, never manipulating its `up` in any way.

Reply:
Okay, if the `up` vector of the generated lights can't be manipulated anyway it's okay to use a default vector.